pandas/core/arrays/datetimes.py

```diff
@@ -2275,6 +2275,19 @@ def to_julian_date(self) -> npt.NDArray[np.float64]:
     # -----------------------------------------------------------------
     # Reductions
 
+    def _reduce(
+        self, name: str, *, skipna: bool = True, keepdims: bool = False, **kwargs
+    ):
+        result = super()._reduce(name, skipna=skipna, keepdims=keepdims, **kwargs)
+        if keepdims and isinstance(result, np.ndarray):
+            if name == "std":
+                from pandas.core.arrays import TimedeltaArray
+
+                return TimedeltaArray._from_sequence(result)
```

Comment:
Does this possibly preserve the resolution of the input? e.g std of a ms resolution datetime array _should_ return ms resolution

Reply:
yes, the resolution is preserved in `nanops.nanstd`.